### PR TITLE
PP-1079 Fix smartpay namespace issue

### PR DESCRIPTION
**Pull request by kakumara**
## WHAT
1. Barclays (unfortunately) has 2 API name spaces (or have had in past). We must use the newer API names space `http://payment.services.adyen.com` everywhere. We were accidentally using a mixture of deprecated and new namespaces.
2. We have been using `charge_id` which the primary key of our charges table as merchant reference. Fixing it to use externalId.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/service/smartpay/SmartpayPaymentProvider.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public StatusMapper getStatusMapper() {
     private Function<AuthorisationGatewayRequest, String> buildSubmitOrderFor() {
         return request -> aSmartpayOrderSubmitRequest()
                 .withMerchantCode(getMerchantCode(request))
-                .withPaymentPlatformReference(request.getChargeId())
```
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/connector/service/smartpay/SmartpayPaymentProvider.java]: Confused why this change doesn't break things?

Comment (kakumara) [reply src/main/java/uk/gov/pay/connector/service/smartpay/SmartpayPaymentProvider.java]: It doesn't. Smartpay (or payment gateways in general) do not care about what we use as merchant reference. They just record it and playback. 
It is for us (and Government department particularly) make it useful when they query about a particular transaction directly with Gateways. Currently GOV departments only aware of our externalID's and it'll be best they could use the same reference everywhere.

Comment (mrlumbu) [reply src/main/java/uk/gov/pay/connector/service/smartpay/SmartpayPaymentProvider.java]: Ok, got it 👍

**Comment by mrlumbu** [comment]
LGTM 👍
